modules/ROOT/pages/tuning-batch-processing.adoc

```diff
@@ -0,0 +1,24 @@
+= Batch Processing
+ifndef::env-site,env-github[]
+include::_attributes.adoc[]
+endif::[]
+
+Mule runtime engine (Mule) possesses the ability to process messages in batches. Within an application, you can initiate a batch job which is a block of code that splits large messages into individual records, performs actions upon each record, then reports on the results and potentially pushes the processed output to other systems or queues.
+
+Consider having 1 million records processed by a batch application with three steps. Several I/O operations occur as Mule processes each record while they move through the job’s phases. The disk characteristics, along with the workload size, play a key role in the performance of the batch job. Mainly, because during the input phase, an in-disk queue is created with the list of records to be processed, which is constantly read and written by the flow resulting in heavy I/O utilization.
+
+Additionally, batch processing requires having enough available memory to process the threads in parallel, which means moving the records from persistent storage into RAM in a fixed size block. The larger your records and their quantity, the more available memory you need for batch processing.
+
+== Batch Block Size
```

Comment:
Because the previous text refers to "records in a fixed-size block, you probably can continue the text here without the subtitle.
I am only just now getting the sense that "batch" = "block of records." This needs to be explicit from the beginning. 
Consider defining how "batch," "message," "block," "records," "job", and "application" all fit together from the start of the topic. If there are "steps" or "phases," define them as well. Maybe something like...:

"You can improve application processing performance by using _batch processing_, which takes large message blocks of code and divides them into individual records that are then processed individually and simultaneously rather than in a long linear manner. The results of these processed records are then batching into a single result and reported or forwarded to other systems or queues.

Successful processing of multiple individual records simultaneously requires significant RAM, because records are moved to RAM from persistent storage to speed processing time. To balance the needs of performance and working memory, the default size for a batch job is 100 records, based on analysis across a set of representative batch use cases with various record sizes. You can change this value by using the following syntax if, after running comparative tests with different values, you find that the optimum batch size for your use case is different: 

[source,xml,linenums]
----
<batch:job jobName="test-batch" blockSize="${batch.block.size}">
----